Tools/jit/_stencils.py

```diff
@@ -189,14 +193,41 @@ def pad(self, alignment: int) -> None:
         self.disassembly.append(f"{offset:x}: {' '.join(['00'] * padding)}")
         self.body.extend([0] * padding)
 
-    def emit_aarch64_trampoline(self, hole: Hole) -> None:
+    def emit_aarch64_trampoline(self, hole: Hole, alignment: int) -> None:
         """Even with the large code model, AArch64 Linux insists on 28-bit jumps."""
-        base = len(self.body)
+        reuse_trampoline = hole.symbol in self.aarch64_trampolines
+        if reuse_trampoline:
+            # Re-use the base address of the previously created trampoline
+            base = self.aarch64_trampolines[hole.symbol]
+        else:
+            self.pad(alignment)
+            base = len(self.body)
         where = slice(hole.offset, hole.offset + 4)
         instruction = int.from_bytes(self.body[where], sys.byteorder)
         instruction &= 0xFC000000
         instruction |= ((base - hole.offset) >> 2) & 0x03FFFFFF
         self.body[where] = instruction.to_bytes(4, sys.byteorder)
+
+        # Fix the disassembly for the branch to call the trampoline
+        bl_instruction = f"{hole.offset:x}: {instruction:x}      bl      {hole.offset:#x} <{hole.symbol}>  // trampoline"
+        self.disassembly = [
+            bl_instruction if line.startswith(f"{hole.offset:x}:") else line
+            for line in self.disassembly
+        ]
+
+        # Remove the relocation once the bl instruction has been fixed
+        relocation_regex = re.compile(
+            rf"{hole.offset:016x}:\s+{hole.kind}\s+_?{hole.symbol}"
+        )
+        self.disassembly = [
+            line2 for line2 in self.disassembly if not relocation_regex.match(line2)
+        ]
+
+        if reuse_trampoline:
+            # There is no need to emit a new trampoline.
+            return
+
+        # Emit a new trampoline only if is not already present in the Stencil
```

Comment:
This sort of says the same thing three different ways. :)

```suggestion
        if reuse_trampoline:
            return
```